### PR TITLE
Firebase set destroy

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

**Review by quantranhong1999 — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
Rebase on latest changes of #518 just to be sure it's still valid

**Comment by Arsnael** [comment]
Rebased on master

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/FirebaseSubscription.scala [hunk 1]
```diff
@@ -93,7 +94,14 @@ case class DeviceClientIdInvalidException(deviceClientId: DeviceClientId, messag
 
 case class TokenInvalidException(message: String) extends IllegalArgumentException(message)
 
-case class UnparsedFirebaseSubscriptionId(id: Id)
+case class UnparsedFirebaseSubscriptionId(id: Id) {
+  def parse: Either[IllegalArgumentException, FirebaseSubscriptionId] = Try(UUID.fromString(id.value))
```
Comment (vttranlina) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/FirebaseSubscription.scala:98]: We can re-use `FirebaseSubscriptionId.liftOrThrow`

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionSetMethodContract.scala [hunk 2]
```diff
@@ -975,4 +978,215 @@ trait FirebaseSubscriptionSetMethodContract {
...
+           |                        "type": "invalidArguments",
+           |                        "description": "invalid is not a FirebaseSubscriptionId: Invalid UUID string: invalid"
+           |                    }
+           |                ]]
+           |            },
+           |            "c1"
+           |        ]
+           |    ]
+           |}""".stripMargin)
+  }
```
Comment (vttranlina) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionSetMethodContract.scala:1191]: Should we add one more test for `destroyShouldNotRevokeIdOfOtherAccount` ?

**Review by vttranlina — APPROVED** [review]

**Pull request merged**